packages/api/src/utils/client/httpClient.ts

```diff
@@ -279,6 +286,14 @@ export class HttpClient implements IHttpClient {
       if (bearerToken && headers["Authorization"] === undefined) {
         headers["Authorization"] = `Bearer ${bearerToken}`;
       }
+      if (url.username || url.password) {
+        if (headers["Authorization"] === undefined) {
+          headers["Authorization"] = `Basic ${toBase64(`${url.username}:${url.password}`)}`;
+        }
```

Comment:
Throw an error if the user sets the `Authorization` header attribute as well as the `username/password` as options.

Reply:
We don't do that for bearer token either, I was think about it but after looking at other http client implementations (e.g. native node http) I figured usually they just have a defined order of precedence.

In our case: explicit header --> bearer header --> basic auth header

As another side note, if we want to throw an error, this can not be implemented here as it would lazily throw it the first time a request is done which is not ideal. We have to throw misconfiguration errors on startup. This is why I also added URL validation in the constructor of the http client, these runtime errors are bad, especially if you use fallback URLs which might not be queried for hours / days and when they do you notice that the URL is invalid.